tempesta_fw/http.c

```diff
@@ -1709,6 +1709,33 @@ tfw_http_add_x_forwarded_for(TfwHttpMsg *hm)
 	return r;
 }
 
+static int
+tfw_http_add_loc_hdrs(TfwHttpMsg *hm, TfwHttpReq *req)
+{
+	int i, r;
+	TfwLocation *loc = req->location;
+
+	/* TODO #862: req->location must be the full set of options. */
+	if (!loc || !loc->usr_hdrs_sz)
+		loc = req->vhost->loc_dflt;
+	if (!loc || !loc->usr_hdrs_sz)
+		loc = (tfw_vhost_get_default())->loc_dflt;
+	BUG_ON(!loc);
+
+	for (i = 0; i < loc->usr_hdrs_sz; ++i) {
+		r = tfw_http_msg_hdr_add(hm, loc->usr_hdrs[i]);
```

Comment:
Well, the issue isn't clear: it's titled "transform custom HTTP headers", but specifies `hdr_add` only. There is reference to [Nginx's `proxy_set_header`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_set_header) which adds a header in all cases except Host and Connection headers when it actually replaces them. That's absolutely non-flexible. It's better to allow a user to delete and replace any headers as well as add it, so please add header deletion and substitution as well:

        hdr_del <header name>
        hdr_sub <header name> <header value>